glide-core/src/client/value_conversion.rs

```diff
@@ -226,6 +242,26 @@ fn convert_array_to_map(
     Ok(Value::Map(map))
 }
 
+/// Converts a flat array of values to a two-dimensional array, where the inner arrays are two-length arrays representing key-value pairs. Normally a map would be more suitable for these responses, but some commands (eg HRANDFIELD) may return duplicate key-value pairs depending on the command arguments. These duplicated pairs cannot be represented by a map.
+///
+/// `array` is a flat array containing keys at even-positioned elements and their associated values at odd-positioned elements.
+fn convert_flat_array_to_key_value_pairs(array: &[Value]) -> RedisResult<Value> {
```

Comment:
You can use `vec` and simplify it on line 185 too
```suggestion
fn convert_flat_array_to_key_value_pairs(array: Vec<Value>) -> RedisResult<Value> {
```